r/tests/testthat/test-cell_cycle_scoring.R

```diff
@@ -0,0 +1,204 @@
+stub_updateCellSetsThroughApi <- function(cell_sets_object,
+                                          api_url,
+                                          experiment_id,
+                                          cell_set_key,
+                                          auth_JWT,
+                                          append = TRUE) {
+  # empty function to simplify mocking. we test patching independently.
+}
+
+stubbed_cellCycleScoring <- function(req, data) {
+  mockery::stub(cellCycleScoring,
+                "updateCellSetsThroughApi",
+                stub_updateCellSetsThroughApi)
+  cellCycleScoring(req, data)
+}
+
+mock_color_pool <- function(n) {
+  paste0("color_", 1:n)
+}
+
+mock_scdata <- function(phase = "S",
+                        add_cycle_genes = TRUE) {
+  enids <- paste0("ENSG", 1:3000)
+  cell_names <- paste0("cell_", 1:500)
+
+  set.seed(ULTIMATE_SEED)
+  counts <-
+    matrix(
+      rnbinom(
+        length(enids) * length(cell_names),
+        mu = 0.3,
+        size = 5
+      ),
+      ncol = length(cell_names),
+      byrow = TRUE
+    )
+  rownames(counts) <- enids
+  colnames(counts) <- cell_names
+
+  new_names <- enids
+  if (add_cycle_genes) {
+    new_names[1:40] <- Seurat::cc.genes$s.genes[1:40]
+    new_names[41:121] <- Seurat::cc.genes$g2m.genes[1:80]
+  }
+
+  gene_annotations <- data.frame(
+    input = enids,
+    name = new_names,
+    original_name = row.names(counts),
+    row.names = enids
+  )
+
+  pbmc_small <- SeuratObject::CreateSeuratObject(counts = counts)
+
+  if (phase == "S") {
+    pbmc_small@assays$RNA@counts[1:40, ] <- 3
+    pbmc_small@assays$RNA@counts[41:121, ] <- 0
+  } else{
+    pbmc_small@assays$RNA@counts[41:121, ] <- 3
+  }
```

Comment:
this is a bit weird. I understand that you're trying to over express phase-specific genes? in that case, and since you have rownames, you could re-write as  

```suggestion
  if (phase == "S") {
    pbmc_small@assays$RNA@counts[Seurat::cc.genes$s.genes, ] <- 3
    pbmc_small@assays$RNA@counts[Seurat::cc.genes$g2m.genes, ] <- 0
  } else {
    pbmc_small@assays$RNA@counts[Seurat::cc.genes$g2m.genes, ] <- 3
  }
```

Reply:
you could add a comment as well.

Reply:
The counts don't have the symbols used in cc.genes, I will clarify with a comment.